### ISSUE TITLE
Error when publishing app to App Store Connect: `Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib has following signing error(s): code object is not signed at all In architecture: x86_64`

### ISSUE THREAD

**Issue by nilsreichardt** [jerson/flutter-rsa#53]
## Description

When we publish our macOS app to App Store Connect, we get the following email by Apple:

> Dear Developer,
> 
> We identified one or more issues with a recent delivery for your app, "Sharezone - Schulplaner" 1.6.2 (412). Please correct the following issues, then upload again.
>
> ITMS-90238: Invalid Signature - The executable at path Sharezone.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib has following signing error(s): code object is not signed at all In architecture: x86_64 . Refer to the Code Signing and Application Sandboxing Guide at http://developer.apple.com/library/mac/#documentation/Security/Conceptual/CodeSigningGuide/AboutCS/AboutCS.html and Technical Note 2206 at https://developer.apple.com/library/mac/technotes/tn2206/_index.html for more information.
>
> Best regards,
> 
> The App Store Team

I am not sure if someone knows how to fix it. I'm sure we're doing something wrong with signing (see the signing code in "Our code") because when publishing with our MacBook to App Store Connect, everything works.

## Details

* It only happens when we publish our macOS app, our iOS app is fine
* It only happens when publishing with our CD, not publishing with our MacBook via XCode

## "Steps to reproduce"

These are steps that we executed:

1. Followed the docs by Flutter to publish a macOS App: https://docs.flutter.dev/deployment/macos#create-a-build-archive-with-codemagic-cli-tools
2. Publish the app to App Store Connect

## Our code

```sh
APP_NAME=$(find $(pwd) -name "*.app")
PACKAGE_NAME=$(basename "$APP_NAME" .app).pkg
xcrun productbuild --component "$APP_NAME" /Applications/ unsigned.pkg

INSTALLER_CERT_NAME=$(keychain list-certificates \
          | jq '[.[]
            | select(.common_name
            | contains("Mac Developer Installer"))
            | .common_name][0]' \
          | xargs)

xcrun productsign --sign "$INSTALLER_CERT_NAME" unsigned.pkg "$PACKAGE_NAME"
rm -f unsigned.pkg
```

Our GitHub Actions workflow: https://github.com/SharezoneApp/sharezone-app/blob/6c376fe1b7898085b5f0f0af819b6bd942176daf/.github/workflows/safe_app_ci.yml
Our app code: https://github.com/SharezoneApp/sharezone-app/tree/add-alpha-macos/app

## Environment 

* Flutter version: 3.7.12
* `fast_rsa` version: 3.5.7
* Used GitHub Actions with `macos-13` to build and sign the app

**Comment by jerson** [comment]
@nilsreichardt thanks for report this issue, going to take a look next week

**Comment by jerson** [comment]
Hi @nilsreichardt  could you test with this branch, I have updated the binaries built with the latest toolkit

https://github.com/jerson/flutter-rsa/pull/56

**Comment by nilsreichardt** [comment]
Thanks! Unfortunately, it's still the same issue :(

**Comment by nilsreichardt** [comment]
A short update: Meanwhile, I have a similar issue also when building with my local machine and signing with XCode:

> Dear Developer,
> 
> We identified one or more issues with a recent delivery for your app, "Sharezone - Schulplaner" 1.7.3 (445). Please correct the following issues, then upload again.
> 
> ITMS-90238: Invalid Signature - The executable at path Sharezone.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib has following signing error(s): valid on disk /Volumes/workspace/app_data/SWValidationService/mz_15686802414321557788dir/mz_246313557693437340dir/de.codingbrain.sharezone.app.pkg/Payload/Sharezone.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib: does not satisfy its designated Requirement /Volumes/workspace/app_data/SWValidationService/mz_15686802414321557788dir/mz_246313557693437340dir/de.codingbrain.sharezone.app.pkg/Payload/Sharezone.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib: explicit requirement satisfied . Refer to the Code Signing and Application Sandboxing Guide at http://developer.apple.com/library/mac/#documentation/Security/Conceptual/CodeSigningGuide/AboutCS/AboutCS.html and Technical Note 2206 at https://developer.apple.com/library/mac/technotes/tn2206/_index.html for more information.
> 
> Best regards,
> 
> The App Store Team

In https://github.com/SharezoneApp/sharezone-app/pull/630/commits/25b5fd6e6b23904e163d0a14059ab09e81d3bcf5 I removed `fast_rsa` as package from our app and I was able to publish it to [TestFlight](https://testflight.apple.com/join/10tIrtLw).

**Comment by nilsreichardt** [comment]
Another Update: I was also able to publish a fully working version with GitHub Actions when `fast_rsa` is removed.

**Comment by nilsreichardt** [comment]
Okay, I have something reproducible:

1. Clone this repository
2. Navigate to `example` folder
3. Set the App category (e.g. Developer Tools)
4. Run `flutter build macos`
5. In XCode Product Archive
6. Distribute to App Store Connect, choose a unique name

After uploading and waiting ~ 15 minutes, you get this email from Apple:

> Dear Developer,
> 
> We identified one or more issues with a recent delivery for your app, "sharezone_playground" 1.0.2 (3). Please correct the following issues, then upload again.
> 
> ITMS-90238: Invalid Signature - The executable at path [fast_rsa_example.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib](http://fast_rsa_example.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib) has following signing error(s): valid on disk /Volumes/workspace/app_data/SWValidationService/mz_4598802509179499255dir/mz_17001317721199781887dir/de.codingbrain.playground.pkg/Payload/[fast_rsa_example.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib](http://fast_rsa_example.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib): does not satisfy its designated Requirement /Volumes/workspace/app_data/SWValidationService/mz_4598802509179499255dir/mz_17001317721199781887dir/de.codingbrain.playground.pkg/Payload/[fast_rsa_example.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib](http://fast_rsa_example.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib): explicit requirement satisfied . Refer to the Code Signing and Application Sandboxing Guide at http://developer.apple.com/library/mac/#documentation/Security/Conceptual/CodeSigningGuide/AboutCS/AboutCS.html and Technical Note 2206 at https://developer.apple.com/library/mac/technotes/tn2206/_index.html for more information.
> 
> Best regards,
> 
> The App Store Team

**Comment by jerson** [comment]
I see, @nilsreichardt  have you tried signing by using a command like this:

```bash
codesign --force --sign "Developer ID Application: YOUR_TEAM_ID" --preserve-metadata=identifier,entitlements,flags /path_to_Your_App.app/Contents/Frameworks/fast_rsa.framework/Versions/A/Resources/librsa_bridge.dylib

```

**Comment by jerson** [comment]
as a second option could you try by using this branch please

https://github.com/jerson/flutter-rsa/tree/macos-vendor-library

**Comment by jerson** [comment]
Hi @nilsreichardt were you able to test it ?

**Comment by nilsreichardt** [comment]
Somehow, my last and final release processes now over 3 hours in App Store Connect. Not sure what happened there. But my releases before my final release have worked and I think it's fixed 👍

**Comment by jerson** [comment]
cool, thanks, I still need to do some small fixes for flutter test command but other than that is going to be released in 3.6.1

**Comment by nilsreichardt** [comment]
Finally, my final build on App Store Connect proceeded and I was a able to test: it works 😍 Thanks <3
